### DISCUSSION TITLE
Branch Names

### DISCUSSION THREAD

**Discussion by Portnord** [sonic-howl/frc2024#31] [General]
Some of you familiar with git may have noticed we aren't using a typical branching scheme where users create personal branches, for example username/feature. There is a reason for this.

Since we're primarily a teaching project for extremely novice coders, most students are not autonomous enough to do substantial work unsupervised. Many don't have proper equipment to work from home either. So we have two bottlenecks, primarily the amount of mentor attention available and to a much lesser degree the number of computers properly setup.

We often get around that by hot seating. There may be 2 or 3 kids taking turns on the same machine adding to a feature under their personal github credentials while a mentor supervises. Attendance is also not daily so we may often resume work in progress with a different set of hands. So to avoid constantly merging personal branches we just use a general feature/featurename convention.

Comments and suggestions welcome.

**Comment by NathanGrenier** [comment]
> Many don't have proper equipment to work from home either. 

If a student has a laptop or computer, they should to able to follow the readme and install the project.

> Attendance is also not daily so we may often resume work in progress with a different set of hands. So to avoid constantly merging personal branches we just use a general feature/featurename convention.

Good idea, some other prefixes that are quite common / useful might be:
- docs
- wip
- bug

**Comment by Portnord** [comment]
> If a student has a laptop or computer, they should to able to follow the readme and install the project.
> 
The school issues chrome books so everyone has at least that although I don't know what state they are in or if the kids have rights to install.

But you're right and installing it at home is an ideal way to get extra experience coding. It's something to encourage.

> Good idea, some other prefixes that are quite common / useful might be:
> 
> * docs
> * wip
> * bug

Did we write down anything about workflow? From the little we discussed I imagined something simple like:
- feature/* or bug/* -> development -> main
